packages/shared/src/contexts/BootProvider.tsx

```diff
@@ -124,113 +122,106 @@ export const BootDataProvider = ({
     );
   };
 
-  const [cachedBootData, setCachedBootData] =
-    useState<Partial<BootCacheData>>(localBootData);
-  const [lastAppliedChange, setLastAppliedChange] =
-    useState<Partial<BootCacheData>>();
-  const [initialLoad, setInitialLoad] = useState<boolean>(null);
-  const loadedFromCache = !!cachedBootData;
-  const { user, settings, alerts, notifications, squads } =
-    cachedBootData || {};
-  const loggedUser = !!(user && 'providers' in user && user?.id);
+  const initialData = useMemo(() => {
+    if (localBootData) {
+      return localBootData;
+    }
+
+    const boot = getLocalBootData();
+
+    if (!boot) {
+      return null;
+    }
+
+    if (boot?.settings?.theme) {
+      applyTheme(themeModes[boot.settings.theme]);
+    }
+
+    preloadFeedsRef.current({ feeds: boot.feeds, user: boot.user });
+
```

Comment:
Memo without any dependencies, it will execute on every render, just checking if its deliberate?

Reply:
Good point. We can make this a `ref` instead.